zanata-war/src/main/java/org/zanata/rest/editor/MediaTypes.java

```diff
@@ -32,6 +32,11 @@ public String toString() {
     public static final String APPLICATION_ZANATA_LOCALES_JSON =
             APPLICATION_ZANATA_LOCALES + JSON;
 
+    public static final String APPLICATION_ZANATA_SUGGESTIONS =
+            APPLICATION_VND_ZANATA + ".suggestions";
+    public static final String APPLICATION_ZANATA_SUGGESTIONS_JSON =
+            APPLICATION_ZANATA_SUGGESTIONS + JSON;
+
```

Comment:
It will make more sense to return vnd together with xml type. json has no type so really not giving us anything IMHO. I think I've read somewhere mentioning to include API version as media type. Not something we do at the moment. Either way I am fine with whatever we choose. 
----- Original Message -----

> From: "Sean Flanigan" <notifications@github.com>
> To: "zanata/zanata-server" <zanata-server@noreply.github.com>
> Cc: "Patrick Huang" <pahuang@redhat.com>
> Sent: Monday, May 18, 2015 2:29:50 PM
> Subject: Re: [zanata-server] REST endpoint for editor translation memory
> (#816)

> In zanata-war/src/main/java/org/zanata/rest/editor/MediaTypes.java :
> > @@ -32,6 +32,11 @@ public String toString() {
> >      public static final String APPLICATION_ZANATA_LOCALES_JSON =
> >              APPLICATION_ZANATA_LOCALES + JSON;
> >
> > +    public static final String APPLICATION_ZANATA_SUGGESTIONS =
> > +            APPLICATION_VND_ZANATA + ".suggestions";
> > +    public static final String APPLICATION_ZANATA_SUGGESTIONS_JSON =
> > +            APPLICATION_ZANATA_SUGGESTIONS + JSON;
> > +

> Custom media types are kind of an open question in RESTful architecture -
> http://www.infoq.com/news/2010/01/subbu-custom-media-types - old article,
> but still true.

> The problem is that application/json doesn't really tell you anything about
> the semantics of the returned JSON.

> Can't the server say that it @Consumes / @Produces both application/json and
> the custom vnd type?

> —
> Reply to this email directly or view it on GitHub .